src/service.js

```diff
@@ -1,27 +1,61 @@
 import Debug from 'debug';
+import isArray from "lodash/isArray"
+const ACT_SEND = 'send'
+const ACT_BATCH = 'batch'
 
 const debug = Debug('feathers-sendwithus:service');
 
-export default function createService({ api, templateMapper }) {
+export default function createService({ api, templateMapper,batchOpts }) {
+  const {path='/api/v1/send',method='POST',chunkSize=10}=batchOpts || {}
+
   return Object.create({
     setup(app) {
       this.app = app;
     },
 
     create(params) {
-      debug(`create: ${JSON.stringify(params)}`);
-      return templateMapper(params.template)
-        .then(template => new Promise((resolve, reject) => {
-          const data = Object.assign({}, params, { template });
-          api.send(data, (err, result) => {
-            if (err) {
-              reject(err);
-              return;
+      debug(`create: ${JSON.stringify(params)}`)
+      const param_template = (isArray(params))?params[0].template:params.template
+      return templateMapper(param_template)
+        .then(template =>
+          new Promise((resolve, reject) => {
+            const context = { resolve, reject };
+            const data = isArray(params) ? 
+                  params.map(d => ({
+                    body: Object.assign({}, d, { template }),
+                    path,
+                    method,
+                  })
+                }) : Object.assign({}, params, { template });
+        
+            if (isArray(data)) {
+              // TODO: Refactor
+              for (let i = 0; i < data.length; i+=chunkSize) {
+                const chunk = data.slice(i, i + chunkSize);
+                api.batch(chunk, (err, result) => {
```

Comment:
No I decided that abstracting the API call in another method was not worth it. I'd rather move the handling of batch data in another function to keep things small. The refactoring I mention in the comment is replacing the for loop with lodash chunk (since lodash is now a dependency) or perhaps use the async library to do these in series (one after another) rather than all at the same time. I can do this once merged and before publishing.